cmd/pkger/cmds/pack.go

```diff
@@ -143,7 +143,7 @@ func Package(info here.Info, out string, decls parser.Decls) error {
 	}
 
 	name := c.Name
-	if info.Module.Main {
+	if !subdir && info.Module.Main {
```

Comment:
this should be changed to check if the output directory is `main` and if so, force the `main` package name. that is the intent of this code, but it's checking the module and not the directory.

Reply:
> Can also please add tests around this to prevent regression? Thanks.

I don't have any problem doing this, but I don't see any tests of _any_ of the `cmds` package at the moment. Where would you want such tests to live?